voyager-transitions/src/commonMain/kotlin/cafe/adriel/voyager/transitions/ScreenTransition.kt

```diff
@@ -104,8 +167,34 @@ public fun ScreenTransition(
         },
         modifier = modifier
     ) { screen ->
+        if (this.transition.targetState == this.transition.currentState) {
```

Comment:
I suggest changing the condition a little:

```
if (this.transition.targetState == this.transition.currentState && disposeScreenAfterTransitionEnd) {
 ...
}
```

and get rid of `disposeScreenAfterTransitionEnd` check in `LaunchedEffect`